app/models/principal.rb

```diff
@@ -92,7 +92,7 @@ def associate_firm
   end
 
   def match_fca_number
-    unless Lookup::Firm.exists?(fca_number: fca_number)
+    unless fca_authorised_firm?(fca_number)
```

Comment:
What would happen if the FCA call fails? Error page for the user or an error message?